Framework/Algorithms/src/MonteCarloAbsorption.cpp

```diff
@@ -42,12 +45,12 @@ inline double toWavelength(double energy) {
 
 /// Get ith wavelength point for point data
 /// Assumes all checks on sizes have been done before calling
-double getWavelengthPointData(int i, const std::vector<double> &lambdas) {
+double getWavelengthPointData(int i, const HistogramX &lambdas) {
   return lambdas[i];
 }
 
 /// Get ith wavelength point for histogram data
-double getWavelengthHistogramData(int i, const std::vector<double> &lambdas) {
+double getWavelengthHistogramData(int i, const HistogramX &lambdas) {
```

Comment:
You can get rid of these two functions and the whole mess with `LambdaPointProvider lambda` below, I think.